packages/sdk/src/getTransferData/getSourceData.ts

```diff
@@ -40,38 +50,53 @@ export async function getSourceData({
   transferConfig,
   destinationAddress,
   destinationFee,
-  evmSigner,
   polkadot,
   sourceAddress,
+  evmSigner,
 }: GetSourceDataParams): Promise<SourceChainTransferData> {
   const {
     asset,
     destination,
     source: { chain, config },
   } = transferConfig;
+
   const zeroAmount = AssetAmount.fromAsset(asset, {
     amount: 0n,
     decimals: await getDecimals({
       address: destinationAddress,
       chain,
       config,
-      evmSigner,
       polkadot,
     }),
   });
+
+  const feeAsset = polkadot.chain.getBalanceAssetId(
+    config?.fee?.asset || config.asset,
+  );
+
+  const feeCfg = config.fee?.balance.build({
```

Comment:
I don't understand 100% why is this extracted from the function and passed as an argument?

Reply:
If you check the `getDecimals` function you will see that it does: 

```typescript
 config.balance.build({
      address,
      asset: polkadot.chain.getBalanceAssetId(asset || config.asset),
    });
```
Which is wrong. Because balance config is different than `fee?.balance` config.

Reply:
Maybe the way how I did it can be improved. But I decided to do so to not break anything else.

Reply:
ok, I understand now